app/src/main/java/com/rkm/tasky/util/date/DateUtils.kt

```diff
@@ -40,11 +61,39 @@ fun getDayOfTheMonth(date: Long): Int {
     return day.dayOfMonth
 }
 
+fun getReminderTime(date: Long, options: ReminderOptions): Long {
+    val reminder = Instant.fromEpochMilliseconds(date)
+    return reminder.minus(options.time, options.unit, timezone).toEpochMilliseconds()
+}
+
 fun Long.toLocalDateTime(): LocalDateTime {
     return Instant.fromEpochMilliseconds(this)
         .toLocalDateTime(timezone)
 }
 
 fun LocalDateTime.toLong(): Long {
     return this.toInstant(timezone).toEpochMilliseconds()
+}
+
+fun LocalDate.toLong(): Long {
+    return this.atStartOfDayIn(timezone).toEpochMilliseconds()
+}
+
+fun LocalTime.toUiString(): String {
+    return timeFormat.format(this)
+}
+
+fun LocalDate.toUiString(): String {
+    return dateFormat.format(this)
+}
+
+sealed class ReminderOptions(
+    open val time: Int,
+    open val unit: DateTimeUnit
+) {
+    data class TenMinutes(override val time: Int = 10, override val unit: DateTimeUnit = DateTimeUnit.MINUTE): ReminderOptions(time, unit)
+    data class ThirtyMinutes(override val time: Int = 30, override val unit: DateTimeUnit = DateTimeUnit.MINUTE): ReminderOptions(time, unit)
+    data class OneHour(override val time: Int = 1, override val unit: DateTimeUnit = DateTimeUnit.HOUR): ReminderOptions(time, unit)
+    data class SixHour(override val time: Int = 6, override val unit: DateTimeUnit = DateTimeUnit.HOUR): ReminderOptions(time, unit)
+    data class OneDay(override val time: Int = 1, override val unit: DateTimeUnit = DateTimeUnit.DAY): ReminderOptions(time, unit)
```

Comment:
What do you think of using an enum instead here with a single parameter? Here we use Kotlin’s duration type and we don’t have to define the unit type.

```
enum class RemindBeforeDuration(val duration: Duration) {
  TEN_MINUTES(10.minutes),
  THIRTY_MINUTES(30.minutes),
  ONE_HOUR(1.hours),
  SIX_HOURS(6.hours),
  ONE_DAY(1.days)
}
```

Reply:
I didn't know this existed thanks!